pkg/agent/secrets.go

```diff
@@ -52,28 +52,27 @@ func (s *secretSyncCommand) RunWithClientFactory(
 	return s.syncSecret(ctx, client)
 }
 
-func (s *secretSyncCommand) getOps(insertRoot bool) ([]map[string]string, error) {
-	var ops = make([]map[string]string, 0)
+func (s *secretSyncCommand) getOps(insertRoot bool) ([]map[string]interface{}, error) {
+	var ops = make([]map[string]interface{}, 0)
 	// if the secret is empty, this is required.
 	if insertRoot {
-		ops = append(ops, map[string]string{
+		ops = append(ops, map[string]interface{}{
 			"op":    "add",
 			"path":  "/data",
-			"value": "{}",
+			"value": map[string]interface{}{},
```

Comment:
I've changed this for interface{} because it needs to be an object if the value is {}. "{}" won't work.